packages/xstate-react/src/useSelector.ts

```diff
@@ -9,17 +10,32 @@ function defaultCompare<T>(a: T, b: T) {
 export function useSelector<TActor extends ActorRef<any, any>, T>(
   actor: TActor,
   selector: (emitted: SnapshotFrom<TActor>) => T,
+  compare?: (a: T, b: T) => boolean
+): T;
+export function useSelector<TActor extends ActorRef<any, any>, T>(
+  actor: TActor | undefined,
+  selector: (emitted: SnapshotFrom<TActor> | undefined) => T,
+  compare?: (a: T, b: T) => boolean
+): T;
+export function useSelector<TActor extends ActorRef<any, any> | undefined, T>(
+  actor: TActor | undefined,
+  selector: (emitted: SnapshotFrom<TActor> | undefined) => T,
   compare: (a: T, b: T) => boolean = defaultCompare
 ): T {
+  const resolvedActor = actor ?? createEmptyActor();
```

Comment:
The types mention that `selector` might receive `undefined` but since this is implemented using `createEmptyActor` it actually can receive `Snapshot<undefined>`. I think it's going to be easier for people to handle actual `undefined` so I'm going to push out the "fix" for it in a moment